cclib/method/ddec.py

```diff
@@ -631,67 +685,258 @@ def _calculate_w_and_u(self):
+            )
+            for radiusi in range(1, len(self._g[atomi])):
+                self._g[atomi][radiusi] = min(self._g[atomi][radiusi], exp_applied[radiusi - 1])
+            # phi_A^II -- Equation S106 in doi: 10.1039/c6ra04656h
+            phiAII[atomi] = self._integrate_from_radial(
+                [self._g[atomi] - self.rho_wavg[atomi]], [atomi]
+            )
+
+            self._candidates_bigPhi.append([bigphiAII[atomi]])
+            self._candidates_phi.append([phiAII[atomi]])
+
+            phiconverged = True  # when convergence is reached, this is modified to False.
+
+            while phiAII[atomi] <= 0:
+                # Iterative algorithm until convergence
+                # Refer to Figure S3 in doi: 10.1039/c6ra04656h
+                temp = self._integrate_from_radial([numpy.sqrt(self.rho_wavg[atomi])], [atomi])
+                bigphiAII[atomi] = 2 * bigphiAII[atomi] - phiAII[atomi] / temp
+
+                # When Phi is updated, related quantities are updated as well
+                # Refer to S102 in doi: 10.1039/c6ra04656h
+                phiAII[atomi], self._g[atomi] = self._phiaii(
+                    self.rho_wavg[atomi], eta, bigphiAII[atomi], atomi
+                )
+
+                self._candidates_phi[atomi].append(phiAII[atomi])
+                self._candidates_bigPhi[atomi].append(bigphiAII[atomi])
+
+            # lowerbigPhi is largest negative Phi.
+            # upperbigPhi is smallest positive Phi.
+            if phiconverged:
+                self._candidates_phi[atomi] = numpy.array(self._candidates_phi[atomi], dtype=float)
+                self._candidates_bigPhi[atomi] = numpy.array(
+                    self._candidates_bigPhi[atomi], dtype=float
+                )
+                if numpy.count_nonzero(self._candidates_phi[atomi] < 0) > 0:
+                    lower_ind = numpy.where(
+                        self._candidates_phi[atomi]
+                        == self._candidates_phi[atomi][self._candidates_phi[atomi] < 0].max()
+                    )[0][0]
+                    lowerbigPhi = self._candidates_bigPhi[atomi][lower_ind]
+                    lowerphi = self._candidates_phi[atomi][lower_ind]
+                else:  # assign some large negative number
+                    lowerbigPhi = -9999999.0
+                    lowerphi = -9999999.0
+                if numpy.count_nonzero(self._candidates_phi[atomi] > 0) > 0:
+                    upper_ind = numpy.where(
+                        self._candidates_phi[atomi]
+                        == self._candidates_phi[atomi][self._candidates_phi[atomi] > 0].min()
+                    )[0][0]
+                    upperbigPhi = self._candidates_bigPhi[atomi][upper_ind]
+                    upperphi = self._candidates_phi[atomi][upper_ind]
+                else:  # assign some large positive number
+                    upperbigPhi = 9999999.0
+                    upperphi = 9999999.0
+
+            iter = 0
```

Comment:
Can you change this to something else? `iter` is a Python built-in function name.

Reply:
I've changed this (and analogous block in phiai) to `iteration`. Thanks!